lib/super_digit.rb

```diff
@@ -1,15 +1,18 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n) 
+# Proc source: https://www.alexstrick.com/blog/using-rubys-digits-method
+
 def super_digit(n)
-   
-end
-  
+  return n if n <= 9
 
-# Time Complexity - ?
-# Space Complexity - ?
-def refined_super_digit(n, k)
-    
+  digits = n.to_s.chars.map { |digit| digit.to_i }
+  sum = digits.sum
```

Comment:
Note that each of these lines are O(n), and you need to factor in the recursion.